relayer/spy_relayer/src/listener/validation.ts

```diff
@@ -105,35 +111,57 @@ export async function parseAndValidateVaa(
   //   return "VAA is not from a monitored contract.";
   // }
 
-  const isCorrectPayloadType = parsedVaa.payload[0] === 1;
-
-  if (!isCorrectPayloadType) {
-    logger.debug("Specified vaa is not payload type 1.");
-    return "Specified vaa is not payload type 1..";
+  const payloadType = parsedVaa.payload[0];
+  let parsedVaaPayload: any = null;
+  if (payloadType !== 3) {
+    const error = "Specified vaa is not swim payload"
+    logger.debug(error);
+    return error
   }
 
-  let parsedPayload: any = null;
   try {
-    parsedPayload = parseTransferPayload(Buffer.from(parsedVaa.payload));
+    parsedVaaPayload = parseTransferWithArbPayload(Buffer.from(parsedVaa.payload));
   } catch (e) {
     logger.error("Encountered error while parsing vaa payload" + e);
   }
 
-  if (!parsedPayload) {
-    logger.debug("Failed to parse the transfer payload.");
-    return "Could not parse the transfer payload.";
+  if (!parsedVaaPayload) {
+    logger.debug("Failed to parse the transfer with data payload.");
+    return "Could not parse the transfer with data payload.";
+  }
+
+  // Make sure this is specifically a Swim Payload 3
+  let swimPayload: any = null;
+  try {
+    swimPayload = parseSwimPayload(Buffer.from(parsedVaaPayload.extraPayload));
+  } catch (e) {
+    logger.error("Encountered error while parsing swim payload from arbitrary data in payload 3" + e) ;
+    return "Could not parse swim payload";
+  }
+
+  // TODO swim payload version check
```

Comment:
version check should not be a strict boolean. when we release a new swim payload version, we will most likely have to support more than 1 version for a small window.

this logic/check should most likely go into `parseSwimPayload`
if version == 0, then attempt to parse with version 0 schema, else if version == 1 etc. then if none of the versions match then throw an error.